types/jest/index.d.ts

```diff
@@ -199,6 +199,10 @@ declare namespace jest {
 
     type Lifecycle = (fn: ProvidesCallback, timeout?: number) => any;
 
+    interface IClassLike {
```

Comment:
What do you think about calling this interface `FunctionLike`?

Since classes are technically "special functions" and the `describe` implementation obviously works functions as well, I think `FunctionLike` would be better here for the sake of clarity.

I'm thinking about the following scenarios:

```js
const myInstance = new SomeClass();

describe(myInstance.constructor /* I don't know what you would do that but whatever 😅 */, () => {
  // ...
});
```

```js
function doSomething() {
  // ...
}

describe(doSomething, () => {
  // ...
});
```

Reply:
SGTM! Made the change.

I also added `Function` to the allowed type union because some projects don't provide a `name` string in their `Function` typings (since it doesn't exist in IE).